DataFormats/Provenance/interface/EventToProcessBlockIndexes.h

```diff
@@ -0,0 +1,24 @@
+#ifndef DataFormats_Provenance_EventToProcessBlockIndexes_h
+#define DataFormats_Provenance_EventToProcessBlockIndexes_h
+
+/**
+
+\author W. David Dagenhart, created 5 January, 2021
+
+*/
+
+namespace edm {
+
+  class EventToProcessBlockIndexes {
+  public:
+    EventToProcessBlockIndexes() {}
+
+    unsigned int index() const { return index_; }
+    void setIndex(unsigned int value) { index_ = value; }
+
+  private:
+    unsigned int index_ = 0;
```

Comment:
Is index 0 not allowed for actual uses?

Reply:
PoolSource always calls setIndex and explicitly sets the value of index_ for every event and this is saved in EventPrincipal via the function fillEventPrincipal. Zero is an allowed and legal value.  For all other sources, there is still an EventToProcessBlockIndexes object in EventPrincipal. It holds the zero that comes from line 20, but it is never used for any purpose.